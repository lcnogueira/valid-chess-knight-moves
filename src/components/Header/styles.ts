import styled, { css } from 'styled-components'

interface MenuLinksData {
  selected: boolean
}

export const Header = styled.nav`
  ${({ theme: { colors } }) => css`
    margin: 0 auto;
    position: sticky;
    top: 0;
    z-index: 10;
    background-color: ${colors.mainBg};
    border-bottom: 1px solid ${colors.white};
  `}
`

export const MenuLinks = styled.ul`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 6.5rem;
  width: 100%;
`

export const MenuLinksItem = styled.li<MenuLinksData>`
  ${({ selected, theme: { colors } }) => css`
    font-size: 1.5rem;
    display: flex;

    a {
      display: flex;
      align-items: center;
      font-weight: bold;
      padding: 0 5rem;
      color: ${selected ? colors.primary : colors.white};

      :hover {
        color: ${colors.primary};
      }
    }
  `}
`
