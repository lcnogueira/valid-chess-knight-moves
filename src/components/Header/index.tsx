import * as S from './styles'
import Link from 'next/link'
import { useRouter } from 'next/router'

const links = [
  { label: 'Home', href: '/' },
  { label: 'Game', href: '/game' }
]

const Header: React.FC = () => {
  const { pathname } = useRouter()
  return (
    <S.Header>
      <S.MenuLinks>
        {links.map(({ href, label }) => (
          <S.MenuLinksItem key={href} selected={pathname === href}>
            <Link href={href}>
              <a>{label}</a>
            </Link>
          </S.MenuLinksItem>
        ))}
      </S.MenuLinks>
    </S.Header>
  )
}

export default Header
