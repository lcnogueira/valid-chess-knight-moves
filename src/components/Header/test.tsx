import { render } from '@testing-library/react'
import * as Router from 'next/router'
import { NextRouter } from 'next/router'
import Header from '.'

jest.spyOn(Router, 'useRouter').mockReturnValue({
  pathname: '/'
} as NextRouter)

describe('<Header />', () => {
  it('renders correctly', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
