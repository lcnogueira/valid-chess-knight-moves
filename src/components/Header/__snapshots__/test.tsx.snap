// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 128rem;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 10;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 6.5rem;
  width: 100%;
}

.c2 {
  font-size: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  padding: 0 5rem;
  color: #ffa500;
}

.c2 a:hover {
  color: #ffa500;
}

.c3 {
  font-size: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: bold;
  padding: 0 5rem;
  color: #fff;
}

.c3 a:hover {
  color: #ffa500;
}

<div>
  <nav
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          href="/game"
        >
          Game
        </a>
      </li>
    </ul>
  </nav>
</div>
`
